File: frontend/api/address/getAddresses.ts
import { MapAddress } from "@/types";
import client from "../client";

export const getAddresses = (
  cityUid: string,
  name: string,
  house_number: string
) => {
  return client.get<MapAddress[]>(`/addresses/${cityUid}`, {
    params: { name, house_number },
  });
};

getAddresses.queryKey = "getAddresses";

File: backend/internal/usecase/addresses/get_addresses.go
package useCaseAddresses

import (
	"context"
	"log"

	"github.com/uxsnap/fresh_market_shop/backend/internal/entity"
)

func (s *UseCaseAddresses) GetAddresses(ctx context.Context, qFilters entity.QueryFilters) ([]entity.Address, error) {
	log.Printf("ucOrders.GetAddresses; city: %v, name: %v", qFilters.CityUid, qFilters.Name)

	addresses, err := s.addressesRepository.GetAddresses(ctx, qFilters)
	if err != nil {
		log.Printf("failed to get addresses: %v", err)
		return []entity.Address{}, err
	}

	return addresses, nil
}
